pkg/api/pod/warnings_test.go

```diff
@@ -599,11 +600,120 @@ func TestWarnings(t *testing.T) {
 				`spec.affinity.podAntiAffinity.preferredDuringSchedulingIgnoredDuringExecution[1].podAffinityTerm.labelSelector: a null labelSelector results in matching no pod`,
 			},
 		},
+		{
+			name: "no duplicate container ports",
+			template: &api.PodTemplateSpec{Spec: api.PodSpec{
+				Containers: []api.Container{{
+					Name: "foo",
+					Ports: []api.ContainerPort{
+						{ContainerPort: 80, HostPort: 80},
+					},
+				}},
+			}},
+			expected: []string{},
+		},
+
+		{
+			name: "duplicate container ports",
```

Comment:
For failure cases you want to test:

* one container, two ports, same port, same protocol
* one container, two ports, same port, same protocol, same hostPort
* one container, two ports, same port, same protocol, same hostPort, same hostIP
* one container, two ports, one with hostPort set, one without
* one container, two ports, one with hostPort set, one without (order flipped)
* one container, two ports, same hostPort, one with hostIP set, one without
* one container, two ports, same hostPort, one with hostIP set, one without (order flipped)

repeat for two containers, with one port each

Reply:
Added.